File: src/elements/entity/store-value.ts
import { ElementTag } from "../element-tag";
import { XMLSchemaElementAttributes, StringBoolean } from "../../types";
import { ValidationMap } from "../../core/validate";

export class StoreValue extends ElementTag {
    public static readonly TAG = "store-value";
    protected attributes = this.attributes as StoreValueAttributes;

    public convert() {
        return [`${this.attributes["value-field"]}.store();`];
    }

    public getValidation(): ValidationMap {
        return {
            unhandledAttributes: ["do-cache-clear"],
        };
    }
}

interface StoreValueAttributes extends XMLSchemaElementAttributes {
    "value-field": string;
    "do-cache-clear"?: StringBoolean;
}

File: src/main/java/co/uk/stefanpuia/minilang2java/tag/entityop/StoreValue.java
package co.uk.stefanpuia.minilang2java.tag.entityop;

import co.uk.stefanpuia.minilang2java.core.TagInit;
import co.uk.stefanpuia.minilang2java.core.model.MinilangTag;
import co.uk.stefanpuia.minilang2java.core.validate.rule.ImmutableAttributeNameRule;
import co.uk.stefanpuia.minilang2java.core.validate.rule.RuleList;
import co.uk.stefanpuia.minilang2java.tag.entityop.GenericValueMethod.GenericValueMethodAttributes;

@MinilangTag("store-value")
public class StoreValue extends GenericValueMethod<GenericValueMethodAttributes> {

  private final GenericValueMethodAttributes attributes;

  public StoreValue(final TagInit tagInit) {
    super(tagInit);
    this.attributes = new GenericValueMethodAttributes(this);
  }

  @Override
  protected GenericValueMethodAttributes getAttributes() {
    return attributes;
  }

  @Override
  public RuleList getRules() {
    return super.getRules()
        .addRules(ImmutableAttributeNameRule.builder().addUnhandled("do-cache-clear").build());
  }

  @Override
  public String getMethod() {
    return "store";
  }
}
